test/cli_test.c

```diff
@@ -0,0 +1,160 @@
+}
+
+Test(cli, cli_not_finished, .exit_code = EXIT_FAILURE) {
+    char *argv[] = {"./yobemag", "-l", "1abc"};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_gt_than_long_max, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MAX);
+    char buf[n + 2];
+    int c  = snprintf(buf, n + 1, "%li", LONG_MAX);
+    buf[n] = '9';
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_lt_than_long_min, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MIN);
+    char buf[n + 2];
+    int c  = snprintf(buf, n + 1, "%li", LONG_MIN);
+    buf[n] = '9';
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_gt_than_int_max, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MAX);
+    char buf[n + 2];
+    int c = snprintf(buf, n + 1, "%li", LONG_MAX);
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_lt_than_int_min, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MIN);
+    char buf[n + 2];
+    int c = snprintf(buf, n + 1, "%li", LONG_MIN);
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_correct_with_log_lvl, .exit_code = EXIT_SUCCESS) {
+    char *argv[] = {"./yobemag", "-l", "2", "../build/yobemag.gb"};
+    int argc     = 4;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+
+    uint8_t actual_log_lvl   = cli_args.logging_level;
+    uint8_t expected_log_lvl = 2;
+
+    // resolves discarded-qualifiers warning without removing const from CLIArguments
+    char *expected_rom_path = "../build/yobemag.gb";
+    int const len           = snprintf(NULL, 0, "%s", cli_args.rom_path);
+    char actual_rom_path[len + 1];
+    int c = snprintf(actual_rom_path, len + 1, "%s", cli_args.rom_path);
+
+    cr_expect(eq(u8, actual_log_lvl, expected_log_lvl));
+    cr_expect(eq(str, actual_rom_path, expected_rom_path));
+}
+
+Test(cli, cli_correct_no_log_lvl, .exit_code = EXIT_SUCCESS) {
+    char *argv[] = {"./yobemag", "../build/yobemag.gb"};
+    int argc     = 2;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+
+    // resolves discarded-qualifiers warning without removing const from CLIArguments
+    char *expected_rom_path = "../build/yobemag.gb";
+    int const len           = snprintf(NULL, 0, "%s", cli_args.rom_path);
+    char actual_rom_path[len + 1];
+    int c = snprintf(actual_rom_path, len + 1, "%s", cli_args.rom_path);
+
+    cr_expect(eq(str, actual_rom_path, expected_rom_path));
```

Comment:
```suggestion
    cr_expect(eq(str, cli_args.rom_path, expected_rom_path));
```

Maybe I'm missing something and this does not work, but this could get rid of the two `snprintf` above.

Reply:
See comment below